internal/appmetrics/counters.go

```diff
@@ -31,4 +31,30 @@ var (
 		Name: "devices_api_blackbook_request_success_ops_total",
 		Help: "Total successful Blackbook used",
 	})
+	// Chat GPT Metrics
+	OpenAITotalCallsOps = promauto.NewCounter(prometheus.CounterOpts{
+		Name:      "devices_api_openai_chatgpt_ops_total",
+		Help:      "Total number of calls to Open AI ChatGPT",
+		Namespace: "devices_api",
+		Subsystem: "error_codes",
+	})
+	OpenAITotalFailedCallsOps = promauto.NewCounter(prometheus.CounterOpts{
+		Name:      "devices_api_openai_chatgpt_ops_failed_calls_total",
+		Help:      "Total number of failed calls to Open AI ChatGPT",
+		Namespace: "devices_api",
+		Subsystem: "error_codes",
+	})
+	OpenAITotalTokensUsedOps = promauto.NewCounter(prometheus.CounterOpts{
+		Name:      "devices_api_openai_chatgpt_ops_total_token_used",
+		Help:      "Total number of failed calls to Open AI ChatGPT",
+		Namespace: "devices_api",
+		Subsystem: "error_codes",
+	})
+	OpenAIResponseTimeOps = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:      "devices_api_openai_chatgpt_ops_request_duration_seconds",
+		Help:      "Response duration of OpenAI ChatGPT in seconds",
+		Namespace: "devices_api",
+		Subsystem: "error_codes",
+		Buckets:   []float64{0.1, 0.15, 0.2, 0.25, 0.3, 0.5, 0.7, 0.9, 10},
```

Comment:
Given what you've seen, are these reasonable buckets? I feel like I've never seen it take less than 5 seconds. Most of these buckets won't be helpful.

We probably won't get this right the first time.

Reply:
I think there are some built-in Prometheus rules that can replace all of these besides the token count.

Reply:
I've seen it take 11seconds actually

Reply:
Except we decide to give it a cutoff at say 10 and not care about anything after